airbyte-integrations/connectors/source-shopify/unit_tests/integration/api/bulk.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import json
+from datetime import datetime
+from random import randint
+
+from airbyte_cdk.test.mock_http import HttpRequest, HttpResponse
+
+
+def _create_job_url(shop_name: str) -> str:
+    return f"https://{shop_name}.myshopify.com/admin/api/2024-04/graphql.json"
+
+
+def create_job_creation_request(shop_name: str, lower_boundary: datetime, upper_boundary: datetime) -> HttpRequest:
+    outer_query = """mutation {
+                bulkOperationRunQuery(
+                    query: \"\"\"
+                     %INNER_QUERY_TOKEN%
+                    \"\"\"
+                ) {
+                    bulkOperation {
+                        id
+                        status
+                        createdAt
+                    }
+                    userErrors {
+                        field
+                        message
+                    }
+                }
+            }"""
+    inner_query = """{
+  orders(
+    query: \"updated_at:>='%LOWER_BOUNDARY_TOKEN%' AND updated_at:<='%UPPER_BOUNDARY_TOKEN%'\"
+    sortKey: UPDATED_AT
+  ) {
+    edges {
+      node {
+        __typename
+        id
+        metafields {
+          edges {
+            node {
+              __typename
+              id
+              namespace
+              value
+              key
+              description
+              createdAt
+              updatedAt
+              type
+            }
+          }
+        }
+      }
+    }
+  }
+}"""
+    inner_query = inner_query.replace("%LOWER_BOUNDARY_TOKEN%", lower_boundary.isoformat())
+    inner_query = inner_query.replace("%UPPER_BOUNDARY_TOKEN%", upper_boundary.isoformat())
+    outer_query = outer_query.replace("%INNER_QUERY_TOKEN%", inner_query)
```

Comment:
We can also leverage the `stream_slices()` method, to produce 1 slice, or multiple slices, instead of directly injecting the `date-time` str.

Reply:
The same comment as [this](https://github.com/airbytehq/airbyte/pull/38084/files#r1616141758) applies here so I expect the decision we take to be similar. I'll update the thread once we come to an agreement